cmd/tether/net_linux_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build linux
+
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/vishvananda/netlink"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/vmware/vic/lib/config/executor"
+	"github.com/vmware/vic/lib/tether"
+	"github.com/vmware/vic/pkg/vsphere/extraconfig"
+)
+
+// Utility method to add an interface to Mocked
```

Comment:
Yup - not even sure why that's exported. I think it's carried over from tether package when I was exploring how to inherit mock structures (conclusion is that Go makes it hard).
The tether mocking in general needs to be completely revised - my preference is to leave it until vcsim is fleshed out with the ability to "run" workloads and see where we're lacking vs the OS interface that will require.

I've not looked at go-mock recently but it didn't seem sufficient when assembling the tether last year.